### PR TITLE
Winston package: Typescript linting; minor cleanups/refactoring; add some missing exports

### PR DESCRIPTION
**Pull request by DABH**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:
If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/winstonjs/winston
- [ ] Increase the version number in the header if appropriate.
- [X] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

## Notes on this PR

* I have added a tslint.json for the winston package, and accordingly I've made a number of stylistic changes etc. to bring the code into greater compliance with modern Typescript best practices.  There are still a couple issues that should be addressed in the future - hence two rules are excluded in the tslint.json - but the vast majority of linting issues have been resolved.
* Several types including "Config" in the name have been refactored with "ConfigSet" instead.  This is because those types are not a complete configuration for the main type in the package (Winston), but are partial sets of configurations for certain settings.  The Winston source code refers to these as config sets, so this change hopefully also better reflects the original package documentation.
* Moreover, I made a minor refactor where Config is now an interface, and the Winston interface has a Config, increasing the need to differentiate Config from the various ConfigSets.
* Some functions that are exported in the Winston source, such as Winston.config.colorize, were not declared.  I have added the missing functions that I spotted, though coverage may still not be 100%.
* Thank you for your review, I look forward to your comments!

### CODE REVIEW

File: types/winston/index.d.ts [hunk 1]
```diff
@@ -378,32 +376,36 @@ declare namespace winston {
...
         colorize?: boolean;
         colors?: any;
         prettyPrint?: boolean;
-        timestamp?: (Function | boolean);
         showLevel?: boolean;
         label?: string;
         depth?: number;
-        stringify?: Function;
+
+        // tslint:disable-next-line
```
Comment (andy-ms) [inline types/winston/index.d.ts]: Why disabled here? Presumably you are disabling `prefer-method-signature` -- write this line like the next line: `timestamp?(): any | boolean;`.

Comment (DABH) [inline types/winston/index.d.ts]: Thanks, I fixed this per your suggestion (yes, was trying to avoid prefer-method-signature)!

**Comment by DABH** [comment]
Updated this PR to fix style issue and to reflect new path for base tslint config file (so Travis passes again now)

**Comment by DABH** [comment]
Should label on this be changed to Awaiting reviewer feedback?  Unless there are further revisions requested?

**Comment by andy-ms** [comment]
Nope, it's good.

**Pull request merged**